dds/DCPS/security/Utility.h

```diff
@@ -1,32 +1,28 @@
 /*
-*
-*
-* Distributed under the OpenDDS License.
-* See: http://www.OpenDDS.org/license.html
-*/
+ * Distributed under the OpenDDS License.
+ * See: http://www.OpenDDS.org/license.html
+ */
 
 #ifndef OPENDDS_DCPS_SECURITY_UTILITY_H
 #define OPENDDS_DCPS_SECURITY_UTILITY_H
 
-#include "OpenDDS_Security_Export.h"
-#include "TokenReader.h"
-
+#include <dds/DCPS/dcps_export.h>
 #include <dds/DCPS/SequenceIterator.h>
 #include <dds/DCPS/RcObject.h>
 #include <dds/Versioned_Namespace.h>
 
 #include <dds/DdsSecurityCoreC.h>
 
-#if !defined (ACE_LACKS_PRAGMA_ONCE)
-#pragma once
-#endif /* ACE_LACKS_PRAGMA_ONCE */
+#ifndef ACE_LACKS_PRAGMA_ONCE
+#  pragma once
+#endif
 
 OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 
 namespace OpenDDS {
 namespace Security {
 
-class OpenDDS_Security_Export Utility : public DCPS::RcObject {
+class OpenDDS_Dcps_Export Utility : public DCPS::RcObject {
```

Comment:
Is this right? Is it moving from OpenDDS_Security to OpenDDS_Dcps?

Reply:
That seems odd... static builds don't have `Export` macros expand to anything.  Does this mean the CMake static build is significantly different from the MPC static build?

Reply:
Yes, it's very odd. For the most part the CMake should follow MPC's lead, but the new automatic [`StaticIncludes.h`](https://github.com/iguessthislldo/OpenDDS/blob/igtd/cmake-build/dds/DCPS/StaticIncludes.h) is probably the biggest difference that's relevant. It might have a hand in this since it looks like I did this not too long after. I will try to look at it in detail again tomorrow and see if I can see exactly why this was apparently necessary and if there's a better solution.

Reply:
The command line you copied above doesn't link to libOpenDDS_Security.  So with the new code in StaticIncludes, the application has an undefined symbol reference to something in libOpenDDS_Security.

Reply:
Yes, it shouldn't have the initializer in the first place. I think I figured it out though:

- There's probably many ways to get there depending on include order, but somehow [`security/Utility.h` gets included in `security/framework/SecurityConfig.h`](https://github.com/iguessthislldo/OpenDDS/blob/d2767ceb20bf8a6c0396195df6829f23d9ddfc06/dds/DCPS/security/framework/SecurityConfig.h#L22).
- `Utility.h` includes the security export header which sets [`OPENDDS_SECURITY_HAS_DLL` to 0](https://github.com/iguessthislldo/OpenDDS/blob/d2767ceb20bf8a6c0396195df6829f23d9ddfc06/dds/DCPS/security/OpenDDS_Security_Export.h#L12).
- If `StaticIncludes.h` is included first after that, this looks the same as if someone linked the security library ([this is where that's done in CMake](https://github.com/iguessthislldo/OpenDDS/blob/d2767ceb20bf8a6c0396195df6829f23d9ddfc06/cmake/opendds_build_helpers.cmake#L50-L52)), so [it includes the initialization header](https://github.com/iguessthislldo/OpenDDS/blob/d2767ceb20bf8a6c0396195df6829f23d9ddfc06/dds/DCPS/StaticIncludes.h#L38-L40).

This had to be what I figured out last time. I chose to reuse the `*_HAS_DLL` to control `StaticIncludes` because it would also allow overriding `ACE_AS_STATIC_LIBS` in the individual export files without any changes to them. However I guess that depended on there not being a publicly visible include of an export header for a library that's not linked.

Having `Utility` be part of Security really doesn't make any sense as it's part of the interface offered by the security framework. Otherwise there's not really a much of a point for there to be a separate `Utility` and `UtilityImpl` classes. So I do think `Utility` should be moved from Security to Dcps, except more properly. Looking at the commit again where I did this I think it was a quick fix and I meant to follow up on it and get feedback, but forgot about it.